core/mock/mock.go

```diff
@@ -9,28 +11,31 @@ import (
 	blockservice "github.com/ipfs/go-ipfs/blockservice"
 	commands "github.com/ipfs/go-ipfs/commands"
 	core "github.com/ipfs/go-ipfs/core"
-	"github.com/ipfs/go-ipfs/exchange/offline"
+	bitswap "github.com/ipfs/go-ipfs/exchange/bitswap"
+	bsnet "github.com/ipfs/go-ipfs/exchange/bitswap/network"
 	mdag "github.com/ipfs/go-ipfs/merkledag"
 	nsys "github.com/ipfs/go-ipfs/namesys"
 	mocknet "github.com/ipfs/go-ipfs/p2p/net/mock"
-	peer "github.com/ipfs/go-ipfs/p2p/peer"
 	path "github.com/ipfs/go-ipfs/path"
 	pin "github.com/ipfs/go-ipfs/pin"
 	"github.com/ipfs/go-ipfs/repo"
 	config "github.com/ipfs/go-ipfs/repo/config"
-	offrt "github.com/ipfs/go-ipfs/routing/offline"
+	dht "github.com/ipfs/go-ipfs/routing/dht"
 	ds2 "github.com/ipfs/go-ipfs/util/datastore2"
 	testutil "github.com/ipfs/go-ipfs/util/testutil"
 )
 
-// TODO this is super sketch. Deprecate and initialize one that shares code
-// with the actual core constructor. Lots of fields aren't initialized.
-// "This is as good as broken." --- is it?
-
 // NewMockNode constructs an IpfsNode for use in tests.
 func NewMockNode() (*core.IpfsNode, error) {
 	ctx := context.Background()
 
+	// effectively offline, only peer in its network
+	return NewMockNodeOnNet(ctx, mocknet.New(ctx))
+}
+
+// TODO: shrink this function as much as possible, moving logic into NewNode
+func NewMockNodeOnNet(ctx context.Context, mn mocknet.Mocknet) (*core.IpfsNode, error) {
```

Comment:
Since it proved difficult to combine mocknet and the node constructor i was working on, i have this method here as a bit of a shim until i figure out the right way to do it.